Libraries/DirectMLX.h

```diff
@@ -4311,6 +4365,20 @@ namespace dml
 
                         desc.intermediateEdges.push_back(intermediateEdge);
                     }
+                    else if (inputNode.type == NodeType::Constant)
+                    {
+                        // Operator nodes and constant nodes are merged into a single node array
+                        // with all operator nodes appearing before all constant nodes. The node index 
+                        // of the constant node within DML_GRAPH_DESC is the number of operator nodes plus 
+                        // the constant index.
+                        DML_INTERMEDIATE_GRAPH_EDGE_DESC intermediateEdge = {};
+                        intermediateEdge.FromNodeIndex = static_cast<uint32_t>(m_operatorNodes.size()) + inputNode.index;
```

Comment:
Rather than bury this inside an `if` branch, to reduce potential brittleness later (if things get reordered, or another node type is added later), it could be clearer to hoist this up near line 4323. e.g.

`const uint32_t  baseConstantNodeIndex = static_cast<uint32_t>(m_operatorNodes.size());`